lanelet2_python/python_api/io.cpp

```diff
@@ -1,10 +1,40 @@
 #include <lanelet2_io/Io.h>
 
-#include <boost/python.hpp>
+#include "lanelet2_python/internal/converter.h"
 
 namespace py = boost::python;
 using namespace lanelet;
 
+struct DictToConfigurationConverter {
+  DictToConfigurationConverter() { py::converter::registry::push_back(&convertible, &construct, py::type_id<io::Configuration>()); }
+  static void* convertible(PyObject* obj) {
+    if (!PyDict_CheckExact(obj)) {  // NOLINT
+      return nullptr;
+    }
+    return obj;
+  }
+  static void construct(PyObject* obj, py::converter::rvalue_from_python_stage1_data* data) {
+    py::dict d(py::borrowed(obj));
+    py::list keys = d.keys();
+    py::list values = d.values();
+    io::Configuration attributes;
+    for (auto i = 0u; i < py::len(keys); ++i) {
+      std::string key = py::extract<std::string>(keys[i]);
+      if (values[i].operator boost::python::api::bool_type()) {
```

Comment:
Ok, I used a proper way to check for boolean type as described here:
https://www.boost.org/doc/libs/1_57_0/libs/python/doc/v2/extract.html#extract-spec

Reply:
I think that works in the same way. The docs you referenced says about `check`: `false only if no conversion from the stored pointer to T is available`. But since there is a conversion available (e.g. by converting str to bool), this will still return true for non-boolean types.

I think if you really want this to work, you would have to call `isinstance` in some way. But this really seems to me like a lot of effort compared to the solution I proposed. I know its not pretty, but it's working over at the `Attribute` implementation and I heard of no complaints so far.

Reply:
@poggenhans I experimented with this a bit and it works very well. The value has to be something that can be converted into a valid `true` or `false`. Looking at the `TEST(OsmFile, readWriteJSON)` from `TestOsmFile.cpp` and changing the values for `upload`. 

The test asserts
```
EXPECT_EQ(std::string(upload_attribute.name()), "upload");
EXPECT_EQ(std::string(upload_attribute.value()), "true");
```

The following cases pass:
1.  {"josm_upload", true} is the base case
2.  {"josm_upload", "true"}
3. {"josm_upload", 1}

The following values fail the test because the default `false` is used:

1. {"josm_upload", "junk"}
2. {"josm_upload", 1.23}

The following values fail the test because they represent `false`:

1. {"josm_upload", "false"}
2. {"josm_upload", 0}

Now, moving on to the Python side. In addition to `"josm_upload": True`, you can also use

1. "josm_upload": "True"
2. "josm_upload": "true"
3. "josm_upload": 1
4. "josm_upload": "False"
5. "josm_upload": "false"
6. "josm_upload": 0

If you put any other type (e.g., double or object) it will complain that no conversion to string.

It is perfect, in my opinion. Let's leave it.

Reply:
I think we are talking about two separate problems. My concern is that in the future, someone will add a configuration option that expects an `int`. With your python converter, I would expect that this int would always be interpreted as a `bool`, since the `int`->`bool` conversion is possible and so the actual value of the int would be lost in the process of converting it to a `lanelet::Attribute`.

Reply:
Ok, changed it to always go via `string`. The only cost for this is that one cannot do
```
params = {
               "josm_upload": True,
```
in Python, because there is no known boost::python conversion from bool->string. So, the users just do
```
params = {
               "josm_upload": "true",
```
which is not a biggie.

Reply:
Ok, thanks. It's not ideal, I know. But that's probably a task for a new PR, one that also solves it for the Attribute python converter.